sentry-ruby/lib/sentry/transaction.rb

```diff
@@ -205,6 +234,34 @@ def generate_transaction_description
       result
     end
 
+    def populate_head_baggage
+      sentry_items = {}
+      sentry_items["trace_id"] = trace_id
+
+      # TODO-neel filter for high cardinality / low quality tx source here
+      sentry_items["transaction"] = name
+      sentry_items["sample_rate"] = effective_sample_rate&.to_s
+
+      configuration = Sentry.configuration
```

Comment:
If configuration is needed in these new methods, we can use `@configuration` and remove its todo comment.

Reply:
we should decide on a way going forward, I'm fine with using the global since that's what python does too. But if you think having the local variable is cleaner, we can go with that too. What do you prefer?

Reply:
I'm keen to reduce global state because it caused me some problems when taking over `sentry-raven`, especially with its tests.

Reply:
I just added all the variables explicitly in initialize